src/main/java/core/di/factory/ParameterTypeName.java

```diff
@@ -0,0 +1,46 @@
+package core.di.factory;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Parameter;
+import java.util.Arrays;
+
+public class ParameterTypeName {
+    private final String name;
+    private final Parameter parameter;
+    private final int index;
+    private final Annotation[] annotations;
+
+    public ParameterTypeName(String name, Parameter parameter, int index) {
+        this.name = name;
+        this.parameter = parameter;
+        this.index = index;
+        this.annotations = this.parameter.getDeclaredAnnotations();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Class<?> getType() {
+        return this.parameter.getType();
+    }
+
+    public boolean hasAnnotation(Class annotationClass) {
+        return Arrays.stream(this.annotations)
+                .filter(annotation -> annotation.equals(annotationClass))
```

Comment:
아래와 같이 단순화 할 수 있지 않을까요?
```suggestion
                .anyMatch(annotationClass::equals);
```